core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -728,7 +728,13 @@ final class Stream[+F[_],+O] private(private val free: FreeC[Algebra[Nothing,Not
    * res0: List[Int] = List(1, 2, 3, 1, 2, 3, 1, 2)
    * }}}
    */
-  def repeat: Stream[F,O] = this ++ repeat
+  def repeat: Stream[F,O] =
```

Comment:
How does this behave differently than `this ++ repeat`? I see that the handling of `Interrupted` is different but I'm not sure of the implications of `pure(())` vs `interruptEventually(s2.get, interrupted)`.

Reply:
Idea here is that repeat, will always produce same chain of stream to be attached, so this does not make sense to seek the `CloseScope` in the code. Essentially shall be safe to exit the stream chain right away.

Reply:
To clarify the interruptEventually will at best end up with `asHandler` instance of `this` and hence we are already at the end of `this` there will not be any relevant `CloseScope` attached in next of `this` evaluation.

Reply:
Could you add some comments that describe why `repeat` isn't simply `this ++ repeat`? That way I don't try to simplify it 6 months from now when I forget about this. ;p